testeng/resources/upgrade-python-requirements.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -e
+
+virtualenv upgrade_venv -q
```

Comment:
Do we need to explicitly delete this virtualenv when done?  Seems like it could interfere with later jobs on the same worker.

Reply:
We don't seem to have any existing jobs that do that but that's a good idea.